server/src/main/java/org/opensearch/index/search/stats/ShardSearchStats.java

```diff
@@ -110,6 +110,9 @@ public void onFailedQueryPhase(SearchContext searchContext) {
                 if (searchContext.shouldUseConcurrentSearch()) {
                     statsHolder.concurrentQueryCurrent.dec();
                     assert statsHolder.concurrentQueryCurrent.count() >= 0;
+                    if (searchContext.searcher().getSlices() != null) {
+                        statsHolder.queryConcurrencyMetric.inc(searchContext.searcher().getSlices().length);
```

Comment:
We are not recording the `statsHolder.concurrentQueryMetric.inc()` here so seems like concurrentQueryCount will have count of request successfully completed. So we should not record the `queryConcurrency` here as well ?

Not in the scope of this PR, but probably we can introduce another metric to keep track of failed request count and other metrics.

Reply:
@sohami I think you're right, `concurrentQueryMetric` only includes the successful queries, so we shouldn't record concurrency here either